### PR TITLE
SOLR-16427: Evaluate and fix errorprone rules - part 3

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16427

builds on https://github.com/apache/solr/pull/1039

Enables the following errorprone rules:
* `MissingOverride` - this is 90% of the changes and simple `@Override` added
* `ClassCanBeStatic` - few classes updated with `static` keyword
* `EqualsGetClass` - this is most of the other changes to use `instanceof`
* `HidingField` - few classes were overriding protected variables when they shouldn't so they were fixed
* `EscapedEntity / UnescapedEntity` - these were some javadoc fixes
* `ReferenceEquality` - ensure that `.equals()` or `Objects.equals()` used for objects
* `ModifiedButNotUsed` - fixed some cases where it was obvious that wrong variable name was used or variable wasn't needed anymore (ie: `solrUrls`)
* `LogicalAssignment` - fixed a few cases where a variable was assigned inside an `if` statement
* `NarrowingCompoundAssignment` - most of this was changing `int` to `long` to prevent implicit narrowing casts

There are an avg of like 4 lines changed per file. Most are <10 lines and a few bigger ones that had some bad patterns. Overall reasonably simple to review.

### CODE REVIEW

**Review by epugh — APPROVED** [review]
I read through commit by commit, and I like the clarity that these changes bring!   I liked how you did each rule as a separate commit.      LGTM.

File: solr/core/src/java/org/apache/solr/util/hll/SerializationUtil.java [hunk 1]
```diff
@@ -137,8 +137,8 @@ public static byte packVersionByte(final int schemaVersion, final int typeOrdina
    *     <ul>
    *       <li>If 'explicit-disabled' is chosen, this value should be <code>0</code>.
    *       <li>If 'auto' is chosen, this value should be <code>63</code>.
-   *       <li>If a cutoff of 2<sup>n</sup> is desired, for <code>0 <= n < 31</code>, this value
-   *           should be <code>n + 1</code>.
+   *       <li>If a cutoff of 2<sup>n</sup> is desired, for <code>{@code 0 <= n < 31}</code>, this
+   *           value should be <code>{@code n + 1}</code>.
```
Comment (madrob) [inline solr/core/src/java/org/apache/solr/util/hll/SerializationUtil.java:141]: Duplicate code tags?

Comment (risdenk) [reply solr/core/src/java/org/apache/solr/util/hll/SerializationUtil.java:141]: so I went back and forth on this a bunch. I need to look at the actual generated javadoc to see what it looks like. Will check it.

Comment (risdenk) [reply solr/core/src/java/org/apache/solr/util/hll/SerializationUtil.java:141]: ok need to fix this based on https://docs.oracle.com/en/java/javase/19/docs/specs/javadoc/doc-comment-spec.html

Comment (risdenk) [reply solr/core/src/java/org/apache/solr/util/hll/SerializationUtil.java]: fixed in 74ffc2219930d742f38dacee4091d7624c0ddcaf

File: solr/modules/hdfs/src/java/org/apache/solr/hdfs/update/HdfsUpdateLog.java [hunk 2]
```diff
@@ -131,24 +131,24 @@ public void init(UpdateHandler uhandler, SolrCore core) {
...
+              hdfsTlogDir,
               id);
         }
         versionInfo.reload();
         return;
       }
     }
 
-    tlogDir = new Path(dataDir, TLOG_NAME);
+    hdfsTlogDir = new Path(dataDir, TLOG_NAME);
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/modules/hdfs/src/java/org/apache/solr/hdfs/update/HdfsUpdateLog.java:142]: @sonatype-lift ignore

**Comment by risdenk** [comment]
rebased on main again just to run tests one more time.

**Pull request merged**